### PR TITLE
Update READMEs to clarify that SLSA generators and builders must be referred by tag

### PR DESCRIPTION
**Pull request by diogoteles08**
Previously, at some points across the different READMEs there were explanations on why it's necessary to refer the SLSA generators and builders using tags in the format @vX.Y.Z, but some of them were incomplete. This change adds a subtopic to the base README that contains all the explanation, and this subtopic is linked on other parts of READMEs that needs the explanation.

This issue was previously discussed with @laurentsimon

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ This repository contains the code, examples and technical design for system desc
 
 - [Roadmap](#roadmap)
 - [Generation of provenance](#generation-of-provenance)
+  - [Refering SLSA builders and generators](#referring-slsa-builders-and-generators)
```
Comment (asraa) [inline README.md:17]:
```suggestion
  -[Referencing SLSA builders and generators](#referencing-slsa-builders-and-generators)
```

File: internal/builders/generic/README.md [hunk 2]
```diff
@@ -80,8 +80,8 @@ provenance:
     base64-subjects: "${{ needs.build.outputs.hashes }}"
 ```
 
-**Note**: Make sure that you reference the generator with a semantic version of the form `@vX.Y.Z`. The build will fail
-if you reference it via a shorter tag like `@vX.Y` or `@vX`.
+**Note**: Make sure that you reference the generator with a semantic version of the form `@vX.Y.Z`. 
+More information [here](/README.md#referring-slsa-builders-and-generators).
```
Comment (asraa) [inline internal/builders/generic/README.md:84]: referencing-slsa-builders-and-generators here and below like above if you make that change!

**Review by asraa — COMMENTED** [review]
Thanks for helping!

File: README.md (hunk 1 above)
Comment (diogoteles08) [reply README.md]: Thanks for the suggestion! I applied it

File: README.md [hunk 3]
```diff
@@ -42,9 +43,13 @@ timeline for completion.
...
 
-**Note**: At present the GitHub Actions provided in this repository as builders and generators **MUST** be referenced by
-a tag that correpsonds to a semantic version of the form `@vX.Y.Z`. The build will fail
-if you reference it via a shorter tag like `@vX.Y` or `@vX` or if you reference it by a tag of a different form (e.g., `@main`).
+### Referencing SLSA builders and generators
+At present, the GitHub Actions provided in this repository as builders and generators **MUST** be referenced 
+by tag in order for the `slsa-verifier` to be able to verify the ref of the trusted builder/generator's 
+reusable workflow. It also needs to be referred as `@vX.Y.Z`, because the build will fail if you reference it via a shorter tag like `@vX.Y` or `@vX`.
+
+This is contrary to the [best practice](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) which recommends referencing by digest, but intentional due to limits in GitHub Actions.
```
Comment (ianlewis) [inline README.md:51]:
```suggestion
This is contrary to the [GitHub best practice for third-party actions](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) which recommends referencing by digest, but intentional due to limits in GitHub Actions.
```

**Review by ianlewis — APPROVED** [review]

**Pull request merged**